src/components/quiz/UserDetailsForm.tsx

```diff
@@ -0,0 +1,59 @@
+import React, { useState } from "react";
+
+interface User {
+  name: string | null;
+  displayPic: string | null;
+}
+
+export const UserDetailsForm: React.FunctionComponent = ({status,setStatus}: any) => {
+
+  const [userDetails, setUserDetails] = useState<User>({
+    name: null,
+    displayPic:null,
+  });
+  
+
+  function onSubmit(event: any) {    
+    event.preventDefault();     
+    setStatus(true);   
+  }
+
+  return (
+    <div>
+      <h2>Enter your details cadet !</h2>
+   
+      <form action="post">
```

Comment:
Uh-oh! HTML forms love to perform their actions as much as they can. You need to set the form's `onSubmit` event to your handler above, and remove `action="post"`.

What's more, to stop it trying to do it's thing, you need to add:
```
event.preventDefault();
```
to the handler above.

```suggestion
      <form onSubmit={onSubmit}>
```